.github/workflows/rust.yml

```diff
@@ -15,13 +15,16 @@ jobs:
     steps:
       - uses: actions/checkout@v3
       - uses: dtolnay/rust-toolchain@1.69.0
-      - uses: astriaorg/buildjet-rust-cache@v2.5.1
+      - uses: Swatinem/rust-cache@v2.6.1
+        with:
+          cache-provider: "buildjet"
       - name: Install buf cli
         uses: bufbuild/buf-setup-action@v1
         with:
           version: "1.17.0"
           github_token: ${{ secrets.GITHUB_TOKEN }}
       - name: Run tests (excluding RPC client, which needs kubernetes)
+        timeout-minutes: 20
```

Comment:
Just in case, separate the actual test from the building phase so that cold caches don't eat into the test budget, so add an extra step with the `--no-run` flag:
```yaml
      - name: Build tests (excluding RPC client, which needs kubernetes)
        run: |
          cargo test --no-run --release --workspace \
          --all-features \
          --all-targets \
          --exclude astria-celestia-jsonrpc-client
      - name: Run tests (excluding RPC client, which needs kubernetes)
        timeout-minutes: 20
        run: |
          cargo test --release --workspace \
          --all-features \
          --all-targets \
          --exclude astria-celestia-jsonrpc-client
```